merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/services/ConstraintAction.java

```diff
@@ -120,145 +121,126 @@ public Map<Constraint, Fallible<?>> getViolations(final PlanId planId, final Opt
-                                                                   .getOptional()
-                                                                   .get()));
+    }
 
-      // Use the constraints that were compiled and ran with results
-      // to filter out the constraintCode map to match
-      final var compileConstraintCode =
-          constraintCode.entrySet().stream().filter(set -> compiledConstraintMap.containsKey(set.getKey())).collect(
-              Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+    // Store the outcome of the constraint run
+    final var requestId = constraintService.createConstraintRuns(
+        new ConstraintRequestConfiguration(planId, simDatasetId, force, userSession.hasuraUserId()),
+        constraintResultMap);
 
-      // Only update the db when constraints were compiled and ran with results.
-      constraintService.createConstraintRuns(
-          compileConstraintCode,
-          compiledConstraintMap,
-          simDatasetId);
+    return Pair.of(requestId, constraintResultMap);
+  }
+
+  /**
+   * Attempt to compile an EDSL Constraint.
+   * @param modelId The mission model id to get activity and resource types from.
+   * @param planId The plan id to get external resource types from.
+   * @param simDatasetId The simulation dataset id to filter external resource types on.
+   * @param constraint The constraint to be compiled.
+   * @return
+   *    On success, return a {@code Fallible<Expression<EDSLConstraintResult>>} containing the compiled constraint code
+   *      in a form that can be evaluated against simulation results.
+   *    On failure, return a {@code Fallible<Error>} in the Failure state containing the compilation error.
+   */
+  private Fallible<Expression<EDSLConstraintResult>, ConstraintsDSLCompilationService.ConstraintsDSLCompilationResult.Error> tryCompileEDSLConstraint(
+      MissionModelId modelId,
+      PlanId planId,
+      SimulationDatasetId simDatasetId,
+      ConstraintRecord constraint
+  ) {
+    final ConstraintsDSLCompilationService.ConstraintsDSLCompilationResult constraintCompilationResult;
+    try {
+      constraintCompilationResult = constraintsDSLCompilationService.compileConstraintsDSL(
+          modelId,
+          Optional.of(planId),
+          Optional.of(simDatasetId),
+          ((ConstraintType.EDSL) constraint.type()).definition()
+      );
+    } catch (MissionModelService.NoSuchMissionModelException | NoSuchPlanException ex) {
+      return Fallible.failure(
+          new ConstraintsDSLCompilationService.ConstraintsDSLCompilationResult.Error(
+              new ArrayList<>() {{
+                add(new ConstraintsCompilationError(
+                    "Constraint '" +constraint.name()+ "' compilation failed:\n " + ex.getMessage(),
+                    ex.toString(),
+                    new ConstraintsCompilationError.CodeLocation(0,0),
+                    ex.toString()));
+              }}
```

Comment:
This is interesting! I'm unfamiliar with this syntax. I noticed you changed it below to `List.of()`, does it make sense to do that here too? What _does_ `new Foo() {{ ...code here... }}` do?

Reply:
It's known as a "Double Brackets Initializer". From my understanding of the syntax, the first set of brackets creates an anonymous inner class (same as when you do `new Thread() {  public void run() {..}}`, and the inner set opens up the constructor for said anonymous class. 

I think the syntax is neat, but sonarQube doesn't like it, hence why I changed the lower instance to `List.of` after much deliberation (since the List isn't meant to be updated, `List.of` is an acceptable substitute). I can change this one too (or change the lower one to go back to a DBI)

Reply:
Fascinating - that's good to know. I think I agree with SonarQube on this one

Reply:
I swapped this instance to `List.of`